x/evm/keeper/statedb.go

```diff
@@ -572,20 +572,20 @@ func (k *Keeper) AddLog(log *ethtypes.Log) {
 	key := log.TxHash
 
 	if len(k.ctx.TxBytes()) > 0 {
-		tx := &ethtypes.Transaction{}
-		if err := tx.UnmarshalBinary(k.ctx.TxBytes()); err != nil {
-			k.Logger(k.ctx).Error(
-				"ethereum tx unmarshaling failed",
-				"error", err,
-			)
-			return
+		tx, err := k.txDecoder(k.ctx.TxBytes())
+		if err != nil {
+			panic(err)
+		}
+		ethTx, ok := tx.GetMsgs()[0].(*types.MsgEthereumTx)
+		if !ok {
+			panic("invalid ethereum tx")
```

Comment:
unsure if we should panic here or just log and return. What do you think?

Reply:
These invariants should be checked already in ante handlers, log and return could end up with a broken state, panic is better I think.

Reply:
And we only have to calculate the tm hash because of hash inconsistency issue, right? So after that one is solved, it can be simplified here.